.github/scripts/regenerate_cancel_redundant_workflow.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+'''
+This file verifies that the workflows that are potentially canceled in our cancel_redundant_workflow.yml
+match the workflows we have running on pull requests (found in .github/workflows). This way, anytime a
+workflow is added or removed, people can be reminded to modify the cancel_redundant_workflow.yml accordingly.
+'''
+
+
+import ruamel.yaml
+from pathlib import Path
+
+
+yaml = ruamel.yaml.YAML()
+yaml.preserve_quotes = True
+yaml.boolean_representation = ['False', 'True']
+yaml.default_flow_style = False
+
+
+workflow_paths = (Path(__file__).parent.parent / 'workflows').rglob('*')
+workflows = []
+for path in workflow_paths:
+    if path.suffix in {'.yml', '.yaml'}:
+        with open(path) as f:
+            data = yaml.load(f)
+            assert 'name' in data, 'Every GHA workflow must have a name.'
+            if 'pull_request' in data['on']:
+                workflows.append(data['name'])
+
+
+with open('.github/workflows/cancel_redundant_workflows.yml', 'r') as f:
+    data = yaml.load(f)
+    # Replace workflows to cancel
+    data['on']['workflow_run']['workflows'] = sorted(workflows)
+
```

Comment:
you can move this out of the `with` block:
```suggestion

# Replace workflows to cancel
data['on']['workflow_run']['workflows'] = sorted(workflows)
```

Reply:
Is moving it out preferable because the file gets closed when it doesn't need to remain open anymore?

Reply:
Yeah exactly; it's just a nit though, doesn't really matter here haha